### ISSUE TITLE
Error starting phabricator container with host-mounted folders: File '/opt/bitnami/php/conf/php.ini' does not exists

### ISSUE THREAD

**Issue by kdawg14** [bitnami/bitnami-docker-phabricator#89]
**Description**

Running under Docker Toolbox on Windows, trying to use host-mounted folders as the persistent data volumes, I get the following error when trying to `docker-compose up`: 

`Error executing 'postInstallation': File '/opt/bitnami/php/conf/php.ini' does not exists`.

If I use the default docker-compose.yml from the [README](https://github.com/bitnami/bitnami-docker-phabricator#using-docker-compose), it works fine, but when trying to switch to a docker-compose.yml based off [the other one from the README](https://github.com/bitnami/bitnami-docker-phabricator#mount-host-directories-as-data-volumes-with-docker-compose), that's when I get the above error.

I am not overly skilled with docker, so it's entirely possible that I'm doing something wrong.  I did notice a similar issue that someone reported on another [bitnami repo](https://github.com/bitnami/bitnami-docker-suitecrm/issues/30), but I don't really have the wherewithal to troubleshoot it.

Thanks in advance for any help that you can give, and I appreciate you providing this repo!

**Steps to reproduce the issue:**

1. Using the following docker-compose.yml, run `docker-compose up`
```
version: '2'

services:
  mariadb:
    image: 'bitnami/mariadb:latest'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - /D_DRIVE/docker/storage/mariadb:/bitnami
  phabricator:
    image: bitnami/phabricator:latest
    depends_on:
      - mariadb
    ports:
      - '80:80'
      - '443:443'
    environment:
      - PHABRICATOR_HOST=192.168.99.100
    volumes:
      - /D_DRIVE/docker/storage/phabricator:/bitnami
```
That will result in the output that I pasted in below from the phabricator container.  D_DRIVE is a shared folder in my VirtualBox docker machine, and the folders do have data placed in them as a result of running the docker containers, so I don't believe that there are any file access concerns there.

**Describe the results you received:**

Output from the phabricator container: 
```Welcome to the Bitnami phabricator container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-phabricator
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-phabricator/issues
Send us your feedback at containers@bitnami.com
nami    INFO  Initializing apache
apache  INFO  ==> Reconfiguring PID file location...
nami    INFO  apache successfully initialized
nami    INFO  Initializing phabricator
phabric INFO  Creating Phabricator system users...
phabric INFO  Creating Phabricator temporary folders...
phabric INFO  Configuring Apache settings...
phabric INFO  Configuring PHP settings...
Error executing 'postInstallation': File '/opt/bitnami/php/conf/php.ini' does not exists
```

**Describe the results you expected:**
To have it work the same way as the non-host-mounted volumes.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:30:30 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:51:55 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

- Output of `docker info`:

```
Containers: 8
 Running: 0
 Paused: 0
 Stopped: 8
Images: 47
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 154
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.4.74-boot2docker
Operating System: Boot2Docker 17.06.0-ce (TCL 7.2); HEAD : 0672754 - Thu Jun 29 00:06:31 UTC 2017
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.955GiB
Name: default
ID: YAWK:VULS:3IZU:XCIU:NRN7:J62W:H4YJ:DPGP:BJ6Y:XSCJ:NXMV:76DD
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 28
 System Time: 2017-12-28T18:33:58.350672397Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.14.0, build c7bdf9e3
docker-py version: 2.3.0
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Running on Windows under Docker Toolbox using VirtualBox.

**Comment by chipironcin** [comment]
Hello @kdawg14 
Let me explain how mounted volumes work with the modules initialization and try to figure out if there is something wrong in there.

When PHP module initializes, it will look inside the mounted folder for files. If the mounted folder is empty it will populate it. If not empty it will try to use the files inside.

If you have files in `/bitnami/php` (aka `/D_DRIVE/docker/storage/phabricator/php`) then the PHP module wont do its thing believing it is already configured. Then Phabricator will try to do its thing and won't find the right conf file.

Try cleaning the folder `/D_DRIVE/docker/storage/phabricator/php` or deleting it at all and reissuing the deployment.

**Comment by kdawg14** [comment]
Hi, @chipironcin, thanks for the response, and thanks for that explanation.  I actually do not have a `php` folder inside my `/D_DRIVE/docker/storage/phabricator`.  I have an `apache` folder at that level, but no `php`.  As you suggested, I tried entirely deleting the folders under `/D_DRIVE/docker/storage/` and re-ran `docker-compose up`, but I ended up with the same results.

So, it appears that regardless of whether I have the files there or not, it fails.  I did notice that if I started the container again after it went down after the initial failure, it fails at a different point:

```
phabric INFO  Configuring Apache settings...
Error executing 'postInstallation': ELOOP: too many symbolic links encountered, lstat '/opt/bitnami/phabricator/logs/.htaccess'
```

I don't know if that's relevant to the problem I'm experiencing, but thought it might be at least somewhat important.

**Comment by javsalgar** [comment]
Hi,

Just to isolate the issue. Could you confirm if the same happens with another bitnami image that is not Phabricator. For instance, could you try with WordPress?

**Comment by preachermanx** [comment]
This is also failing in the same way with ubuntu 17.10.

http://storage1.static.itmages.com/i/18/0103/s_1515009565_2692011_ca5902afe9.jpeg

**Comment by javsalgar** [comment]
@preachermanx Thanks for letting us know. Could you please provide more information about the environment and the reproduction steps (if they differ from the one already reported)?

**Comment by preachermanx** [comment]
`             /////////////                preachermanx@holmes-iv 
         /////////////////////            ---------------------- 
      ///////*767////////////////         OS: Pop!_OS 17.10 (Artful Aardvark) x86_64 
    //////7676767676*//////////////       Host: Super Server 0123456789 
   /////76767//7676767//////////////      Kernel: 4.13.0-21-generic 
  /////767676///*76767///////////////     Uptime: 9 days, 4 hours, 36 mins 
 ///////767676///76767.///7676*///////    Packages: 2183 
/////////767676//76767///767676////////   Shell: zsh 5.2 
//////////76767676767////76767/////////   Terminal: /dev/pts/0 
///////////76767676//////7676//////////   CPU: Intel Xeon E5-2603 v4 (6) @ 1.700GHz 
////////////,7676,///////767///////////   GPU: ASPEED Technology, Inc. ASPEED Graphics Family 
/////////////*7676///////76////////////   Memory: 1274MiB / 32059MiB 
///////////////7676//////////////////// 
 ///////////////7676///767////////////                            
  //////////////////////'////////////   
   //////.7676767676767676767,//////    
    /////767676767676767676767/////     
      ///////////////////////////       
         /////////////////////          
             ///////////// 
 
`
`preachermanx@holmes-iv ~  
❯ docker version
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:11:14 2017
 OS/Arch:	linux/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:09:47 2017
  OS/Arch:	linux/amd64
  Experimental:	false
`

**Comment by beohof** [comment]
Hi Guys :),

i have the same issue.

Here are my versions:

`Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:11:19 2017
 OS/Arch:       linux/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:09:54 2017
  OS/Arch:      linux/amd64
  Experimental: false
Containers: 21
 Running: 20
 Paused: 0
 Stopped: 1
Images: 23
Server Version: 17.12.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 228
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-5-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.796GiB
Name: removed
ID: removed
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

docker-compose version 1.17.0, build ac53b73
docker-py version: 2.5.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.1t  3 May 2016
`

**Comment by WhyKayPee** [comment]
+1, with the same issue. Persistence is ok for mariadb and fails for phabricator. 
Cheers to all,

**Comment by WhyKayPee** [comment]
If it can help somebody, while waiting things to get better, I fixed it using this dirty hack,

cp -r /var/lib/docker/volumes/bitnamidockers_phabricator_data/_data/php /home/phabricator/dockers-bitnami/phabricator-data/

**Comment by kdawg14** [comment]
Sorry for the late reply, and thanks to those who have confirmed the issue.

@javsalgar, I tried running WordPress with persistent volumes and it ran successfully, indicating that php was successfully initialized, and I checked my directory and it did have both an `apache` and `php` folder within.

**Comment by tompizmor** [comment]
Hi @kdawg14, @preachermanx, @beohof , @WhyKayPee 

This is just a quick note to let you know that we have spotted the issue and we are working on release a new image that fixes this issue.

**Comment by tompizmor** [comment]
Hi again @kdawg14, @preachermanx, @beohof, @WhyKayPee 

We just released Phabricator `2017.52.0-r1` Docker Image fixing the issue with the host volume for php. 
Pull the latest image and try again. Please clean the host folders to start from scratch.
